torch/distributed/_spmd/api.py

```diff
@@ -153,10 +162,21 @@ def _dtensor_expand(
     for a in flat_args:
         if isinstance(a, torch.Tensor):
             inps.append(a)
-            schemas.append(shard_schema)
-        elif isinstance(a, nn.Module) or isinstance(a, torch.optim.Optimizer):
-            # nn.Module or optimizer placeholder is captured by make_fx but
-            # never used in the graph
+            if id(a) in _placements_override:
```

Comment:
TODO for @mrshenli: investigate if there is a better way to handle these overrides.